parcels/field.py

```diff
@@ -310,8 +310,13 @@ def eval(self, time, x, y):
         return self.units.to_target(value, x, y)
 
     def ccode_eval(self, var, t, x, y):
-        return "temporal_interpolation_linear(%s, %s, %s, %s, %s, %s, &%s)" \
-            % (x, y, "particle->xi", "particle->yi", t, self.name, var)
+        # Casting interp_method to int as easier to pass on in C-code
+        if self.interp_method is 'linear':
+            interp_method = 0
```

Comment:
If we use enum value for the method selection, this here also needs changing to `interp_method = 'LINEAR/NEAREST'`, I think.